tests/karate/src/test/java/common/net/MultiMsgWebSocketClient.java

```diff
@@ -0,0 +1,106 @@
+package common.net;
+
+import com.intuit.karate.Logger;
+import com.intuit.karate.http.WebSocketClient;
+import com.intuit.karate.http.WebSocketOptions;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+/**
+ * A WebSocketClient that can handle multiple received messages
+ */
+public class MultiMsgWebSocketClient extends WebSocketClient implements MessageBuffer {
```

Comment:
semantically, does it make sense to say that the client "is" a MessageBuffer as opposed to "has" a MessageBuffer? 
Do we really need the delegate functions as opposed to having a getter for a MessageBuffer ?

Reply:
The Multi-Message Client stores incoming messages, so I don't see it being a Buffer as a problem.

Also, it is for the karate implementation.
I like the idea to have a test that looks like this :

```
Then def msg1 = client.take()
And def msg2 = client.take()
And def msg3 = client.take()
```

rather than 

```
Then def msg1 = client.getBuffer().take()
And def msg2 = client.getBuffer().take()
And def msg3 = client.getBuffer().take()
```


But, now that I think about it, we could also write 
```
Then def buffer = client.getBuffer()
And def msg1 = buffer.take()
And def msg2 = buffer.take()
And def msg3 = buffer.take()
```